scripts/minIO_import.py

```diff
@@ -0,0 +1,132 @@
+1. Start the SSH tunnel to the minIO server at NERSC:
+    ssh -L 9002:ci07:9002 $USERNAME@login1.berkeley.kbase.us -N
+2. Get the secret key from the minIO server in Rancher and set the environment variable:
+    i. Go to the minIO server in Rancher
+        Stacks → CI → cdm-minio -> ci-core-cdm-minio-1 → Execute Shell → echo $MINIO_ROOT_PASSWORD
+    ii. Set the environment variable in the terminal:
+        export SECRET_KEY='your_secret_key'
+"""
+
+SECRET_KEY = os.environ.get('SECRET_KEY')
+ACCESS_KEY = 'cdm-admin'
+ENDPOINT_URL = 'http://localhost:9002'
+
+SOURCE_DIR = Path('/global/cfs/cdirs/kbase/collections') / 'sourcedata' / 'NCBI' / 'NONE'
+SUFFIX = 'protein.faa.gz'
+BUCKET = 'cdm'
+NUM_THREADS = 128
+
+
+def _get_genome_ids_with_lineage(
+        taxonomy_files: list[str | Path],
+        lineages: list[str]
+) -> list[str]:
+    """
+    Get genome ids with the specified lineage from GTDB taxonomy files (e.g. bac120_taxonomy_r214.tsv).
+    """
+    genome_ids = list()
+
+    for file_path in taxonomy_files:
+        with open(file_path, 'r') as file:
+            for line in file:
+                columns = line.strip().split('\t')
+
+                for lineage in lineages:
+                    if lineage in columns[1]:
+                        # Trim the prefix of the genome id as it is not part of the id in NCBI
+                        # e.g. RS_GCF_000979555.1 -> GCF_000979555.1
+                        genome_ids.append(columns[0][3:])
+
+    return genome_ids
+
+
+def _find_files_with_suffix(
+        directory: Path,
+        suffix: str
+) -> list[str]:
+    """
+    Find files with the specified suffix in the directory and its subdirectories.
+    """
+    matching_files = []
+
+    for root, dirs, files in os.walk(directory):
+        for file in files:
+            if file.endswith(suffix):
+                matching_files.append(os.path.join(root, file))
+
+    return matching_files
+
+
+def _upload_to_s3(
+        genome_id: str,
+        s3_client: boto3.client,
+        source_dir: Path,
+        suffix: str,
+        bucket: str,
+        no_match_genome_ids: list[str],
+        multi_match_genome_ids: list[str]):
+    """
+    Upload the genome files to the specified S3 bucket.
+    """
+    genome_dir = source_dir / genome_id
+    matching_files = _find_files_with_suffix(genome_dir, suffix)
+
+    if len(matching_files) == 1:
+        upload_file = genome_dir / matching_files[0]
+        s3_key = 'NCBI/' + upload_file.name
+        try:
+            # skip uploading if the file already exists in the bucket
+            s3.head_object(Bucket=bucket, Key=s3_key)
+        except s3.exceptions.ClientError:
+            s3_client.upload_file(upload_file, bucket, s3_key)
+    else:
+        multi_match_genome_ids.append(genome_id) if matching_files else no_match_genome_ids.append(genome_id)
+
+
+if __name__ == '__main__':
+    meta_dir = Path('/global/homes/t/tgu/GTDB_meta')
```

Comment:
If you wind up making changes it'd be better if:
```suggestion
def main():
   ...

if __name__ == '__main__':
    main()
```
So all these variable names aren't globals. Not necessary for this PR though

Reply:
👍

Reply:
:+1: